src/net_processing.cpp

```diff
@@ -2439,44 +2439,44 @@ bool ProcessMessages(CNode* pfrom, CConnman& connman, std::atomic<bool>& interru
     //  (4) checksum
     //  (x) data
     //
-    bool fOk = true;
+    bool fMoreWork = false;
 
     if (!pfrom->vRecvGetData.empty())
         ProcessGetData(pfrom, chainparams.GetConsensus(), connman, interruptMsgProc);
 
     // this maintains the order of responses
-    if (!pfrom->vRecvGetData.empty()) return fOk;
+    if (!pfrom->vRecvGetData.empty()) return true;
 
     auto it = pfrom->vRecvMsg.begin();
-    while (!pfrom->fDisconnect && it != pfrom->vRecvMsg.end()) {
+    if (!pfrom->fDisconnect && it != pfrom->vRecvMsg.end()) {
```

Comment:
Need to clarify in the commit message: It was almost always the case that only one message was processed in this while loop. See the break at line 2358.

This commit changes the behavior so that it's _always_ one message processed per-loop. The only messages that show different behavior are the ones that used to "continue" the loop. I think it's perfectly reasonable to skip to the next node for processing in that case.